R/Task_operators.R

```diff
@@ -320,6 +320,18 @@ getTaskData = function(task, subset = NULL, features, target.extra = FALSE, reco
     if (recode.target %nin% c("no", "surv")) {
       res[, tn] = recodeY(res[, tn], type = recode.target, task$task.desc)
     }
+    # first condition checks if 'getTaskData' was called directly, i.e. checks if
+    # the call was from the GlobalEnv
+    # second condition checks which function called 'getTaskData' if call was not
+    # from the GlobalEnv. If cond2 is FALSE, 'getTaskData' was called from
+    # 'subsetTask' in a nested resampling call. In this case we remove x and y
+    # later as we still need it for partitioning
+    if (!identical(parent.frame(n = 1), globalenv()) &&
+        !sys.call(-2) == "subsetTask(.task, .subset)" &&
+        task$task.desc$is.spatial == TRUE) {
+      res$x = NULL
+      res$y = NULL
+    }
   }
```

Comment:
To verify that I understand it correctly: You would prefer to 'remove' `x` and `y` before creating the task, i.e. not having them in a data frame initially but passing them (numeric vectors) via a list in an argument called `spatial` like so?:

```r
task <- makeClassifTask(id = "pathogen_data", data = basque, 
                        target = "diplo01", positive = "1",
                        spatial = list(x = randomvectorA, y = randomvectorB))
```